staging/src/k8s.io/api/admissionregistration/v1alpha1/types.go

```diff
@@ -74,6 +75,64 @@ type ValidatingAdmissionPolicy struct {
+	// Populated by the system.
+	// Read-only.
+	// +optional
+	Status ValidatingAdmissionPolicyStatus `json:"status,omitempty" protobuf:"bytes,3,opt,name=status"`
+}
+
+// ValidatingAdmissionPolicyStatus represents the status of a ValidatingAdmissionPolicy.
+type ValidatingAdmissionPolicyStatus struct {
+	// The generation observed by the controller.
+	// +optional
+	ObservedGeneration int64 `json:"observedGeneration,omitempty" protobuf:"varint,1,opt,name=observedGeneration"`
+
+	// TypeChecking contains warnings about identifiers and their types in the
+	// expressions.
+	// +optional
+	TypeChecking *TypeChecking `json:"typeChecking,omitempty" protobuf:"bytes,2,opt,name=typeChecking"`
+
+	// Conditions contains the latest available observations of a policy's current state.
+	// +optional
+	Conditions []ValidatingAdmissionPolicyCondition `json:"conditions,omitempty" protobuf:"bytes,3,rep,name=conditions"`
+}
+
+// ValidatingAdmissionPolicyCondition describes the state of an admission validation policy at a certain point.
+type ValidatingAdmissionPolicyCondition struct {
+	// Type of deployment condition.
+	Type ValidatingAdmissionPolicyConditionType `json:"type" protobuf:"bytes,1,opt,name=type,casttype=ValidatingAdmissionPolicyConditionType"`
+	// Status of the condition, one of True, False, Unknown.
+	Status api.ConditionStatus `json:"status" protobuf:"bytes,2,opt,name=status,casttype=k8s.io/api/core/v1.ConditionStatus"`
+	// The last time this condition was updated.
+	LastUpdateTime metav1.Time `json:"lastUpdateTime" protobuf:"bytes,3,opt,name=lastUpdateTime"`
+	// Last time the condition transitioned from one status to another.
+	LastTransitionTime metav1.Time `json:"lastTransitionTime" protobuf:"bytes,4,opt,name=lastTransitionTime"`
+	// The reason for the condition's last transition.
+	Reason string `json:"reason" protobuf:"bytes,5,opt,name=reason"`
+	// A human-readable message indicating details about the transition.
+	Message string `json:"message" protobuf:"bytes,6,opt,name=message"`
+}
+
+// ValidatingAdmissionPolicyConditionType is the condition type of an admission validation policy.
+type ValidatingAdmissionPolicyConditionType string
+
+// TypeChecking contains results of type checking the expressions in the
+// ValidatingAdmissionPolicy
+type TypeChecking struct {
+	// ExpressionWarnings contains type checking warnings for each expression.
+	// +optional
+	// +listType=atomic
+	ExpressionWarnings []ExpressionWarning `json:"expressionWarnings,omitempty" protobuf:"bytes,1,rep,name=expressionWarnings"`
+}
+
+// ExpressionWarning is a warning information that targets a specific expression.
+type ExpressionWarning struct {
+	// Expression is the CEL expression that this warning is associated.
+	Expression string `json:"expression" protobuf:"bytes,1,opt,name=expression"`
+	// Warning contains information about the expression in a human-readable form.
+	Warning string `json:"warning" protobuf:"bytes,2,opt,name=warning"`
```

Comment:
It wasn't obvious to me from reading this `Warning` field documentation that the contents may contain multiple warnings separated by newlines and that lines are prefixed by the GVK of object kind that the error relates to.

Reply:
Good point. Adding doc.